server/status_test.go

```diff
@@ -118,9 +130,96 @@ func TestStatusJson(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
+			t.Log(string(body))
 			if matches, err := regexp.Match(spec.expected, body); !matches || err != nil {
 				t.Errorf("expected match: %t; err nil: %s", matches, err)
 			}
 		}
 	}
 }
+
+// TestStatusGossipJson ensures that the output requrest for the full gossip
+// info contains the required fields.
+func TestStatusGossipJson(t *testing.T) {
+	s := StartTestServer(t)
+	defer s.Stop()
+
+	type prefixedInfo struct {
+		Key string                 `json:"Key"`
+		Val []storage.PrefixConfig `json:"Val"`
+	}
+
+	type rangeDescriptorInfo struct {
+		Key string                `json:"Key"`
+		Val proto.RangeDescriptor `json:"Val"`
+	}
+
+	type keyValueStringPair struct {
+		Key string `json:"Key"`
+		Val string `json:"Val"`
+	}
+
+	type keyValuePair struct {
+		Key string                 `json:"Key"`
+		Val map[string]interface{} `json:"Val"`
+	}
+
+	type infos struct {
```

Comment:
We're serializing an unexported structure in the gossip system; perhaps the status endpoint should return a different structure which is exported?